validator/src/admin_rpc_service.rs

```diff
@@ -596,29 +664,14 @@ impl AdminRpc for AdminRpcImpl {
         })
     }
 
-    fn set_public_tpu_address(
-        &self,
-        meta: Self::Metadata,
-        public_tpu_addr: SocketAddr,
```

Comment:
alternative is this function takes both tpu and tpu-forwards argument; e.g. as `Option<SocketAddr>`

Reply:
> e.g. as `Option<SocketAddr>`

That was my initial approach. However, the resulting code was not very clean and readable. If we don't add a special function to ClusterInfo, there might be a situation where the first request (tpu) is successful, but the second one (tpu_forwards) is not. Then, the `AdminRpc::set_public...` call would return an error, which in turn leads to an unclear state at the CLI level. In short, it complicates the code. The current implementation turned out to be quite elegant and consistent from my point of view. Additionally, it can be further developed for public-rpc addresses in the future.
Based on this, I suggest keeping the structure of the function signatures as they are.